lib/react_client.dart

```diff
@@ -493,20 +509,25 @@ final ReactDartInteropStatics2 _dartInteropStatics2 = (() {
         component.componentWillReceiveProps(new JsBackedMap.backedBy(jsNextProps));
       });
 
-  void _updatePropsAndStateWithJs(Component2 component, JsMap props, JsMap state) {
+  void _updatePropsAndStateWithJs(Component2 component, JsMap props, JsMap state, [dynamic context]) {
     component
       ..props = new JsBackedMap.backedBy(props)
-      ..state = new JsBackedMap.backedBy(state);
+      ..state = new JsBackedMap.backedBy(state)
+      ..context = _unjsifyNewContext(context);
   }
 
-  bool handleShouldComponentUpdate(Component2 component, JsMap jsNextProps, JsMap jsNextState) => zone.run(() {
-        final value = component.shouldComponentUpdate(
+  bool handleShouldComponentUpdate(Component2 component, JsMap jsNextProps, JsMap jsNextState, dynamic jsNextContext) =>
```

Comment:
it doesn't need to be because on the JS side (where this is called in `_dart_helpers.js` it is always an argument so the value/whatever it may be will always be included, however i can make it optional just to match the dart api and just in case we ever change the JS side to not include it...